general/mm.hpp

```diff
@@ -112,6 +112,23 @@ class mm
       return MM().Known(a);
    }
 
+   // **************************************************************************
+   template<class T>
+   static inline void RegisterHostPtr(T * ptr_host, const size_t size)
+   {
+     MM().Insert(ptr_host, size*sizeof(T));
+   }
+
+   // **************************************************************************
+   template<class T>
+   static void RegisterHostAndDevicePtr(T * ptr_host, T * ptr_device, const size_t size)
+   {
+     MM().Insert(ptr_host, size*sizeof(T));
+     mm::memory &base = MM().maps.memories.at(ptr_host);
+     base.d_ptr = ptr_device;
+     base.host = false;
```

Comment:
We should add the ability to the user to decide where the starting point is.
Setting this boolean to `false` means it is considered to be on the device, which might not be the case.